### ISSUE TITLE
Footnotes shouldn't display code

### ISSUE THREAD

**Issue by vr8hub** [typora/typora-issues#1366]
Typora .9.9.13.6 on OSX 10.12.6. (I looked and couldn't find this, although it relates to #1317).

I'm not sure why footnote text displays with the code instead of just the number/reference, but in a WYSIWYG it's a bit of an anachronism. I would much prefer they display as they would on a web page, i.e. either with a superscripted number (or reference) and the text, or the reference and a period and the text, etc.

<sup>1</sup>This is footnote text, or
1. This is footnote text, but please not this

[^1]:  This is footnote text.

**Comment by abnerlee** [comment]
relates  #1317 and https://github.com/typora/typora-issues/issues/1163

We want to allow users to easily find and modify the footnote id, but they can't do this if all footnote id are hidden and displayed as number.

Also, user may not be able to add footnote and footnote definition in the same time, so how to put number of footnote ids which are not used, or footnote which are not defined would also be an issue.

**Comment by vr8hub** [comment]
You keep mixing WYSIWYG mode with editing the source code. We're using a WYSIWYG editor because we want WYSIWYG!
This was the whole point of #1317 that you continue to argue against. There are numerous issues entered here that all point to the same thing — clean up WYSIWYG mode and get rid of the anomalies.

If we need to edit the code, we either switch to source mode (that's what it's for), or, if you followed through with the #1317 implementation, we would switch focus to the footnote and be able to edit it (if we had it turned on, but even if we didn't, it's on the View menu or a hotkey away). Either way, there's no reason to sully the WYSIWYG mode with source code.

I also didn't say anything about hiding footnote ID's as numbers. I displayed numbers because that's what I use for the ID's. If we used characters (asterisks, etc.), then that is what would display, but the principal is exactly the same.

If we didn't enter the footnote definition at the same time, then there would be nothing to display at the bottom, and the whole issue is moot. I'm talking about when there are footnotes to display, and when there are, then they have ID's, be they numbers or characters or whatever.

On many of these issues, you're arguing for a theoretical user, when real users are requesting that it be made consistent. In this as in science, practice trumps theory. Go with the real users.
